DatePicker V1

AaronPlave opened the pull request:
Initial Date Picker implementation. Closes #44 

Features:
- Input date and time using the text field
- Select day by clicking on the calendar button to open the calendar popover from react-day-picker
- Switch months and years from within calendar picker (does not affect date until a day is selected)
- `formatString` string prop to specify date format, see https://date-fns.org/docs/format for docs. Defaults to "ISO" which is not part of date-fns since normal ISO string cannot be specified using normal date string.
- `min` and `max` date props to bound the date selection
- `minWidth` number prop to specify min width of text input, useful for sizing based off the format
- `errorString` string prop to pass in an external validation that will be shown if all other internal validations first pass
- `renderHeader` and `renderFooter` function props that take `setDateState` and `setOpen` as args. Useful for adding additional content above and below the calendar content in the popover.
- `utc` boolean prop (defaults to true) to specify if the time should be in UTC or local time.
- `adornment` React.ReactNode prop for passing in additional content to the left of the calendar button on the right side of the input.
- Validation including basic date validity, date adherence to optional min, max, and required props
<img width="331" alt="image" src="https://github.com/nasa-jpl/react-stellar/assets/4419607/043fe94c-62dc-4a35-abc6-b5221176cd74">

AaronPlave committed:
DatePicker MVP

AaronPlave committed:
Clean

AaronPlave committed:
Clean

AaronPlave committed:
Clean

AaronPlave committed:
Fixes

AaronPlave committed:
Ensure popover closes when an external button is clicked

AaronPlave committed:
Fix

duranb reviewed (approved):
Looks great! Did we also want to expose a `renderDay` prop to pass through to `DayPicker` so that users can provide custom day components?

AaronPlave commented:
> Looks great! Did we also want to expose a `renderDay` prop to pass through to `DayPicker` so that users can provide custom day components?

Great suggestion. I've added a `renderDay` prop which is passed the `DayContentProps` from DayPicker. Added a story where it uses `renderDay` to show day of year below the date.

AaronPlave committed:
Add renderDay prop to picker and add a DOY story

AaronPlave committed:
Bump version

duranb commented:
perfect! :shipit:
